pkg/client/fake/client.go

```diff
@@ -813,8 +829,39 @@ func (c *fakeClient) patch(obj client.Object, patch client.Patch, opts ...client
 		return err
 	}
 
+	o, err := c.tracker.Get(gvr, accessor.GetNamespace(), accessor.GetName())
+	if err != nil {
+		return err
+	}
+	oldObj, err := meta.Accessor(o)
```

Comment:
```suggestion
	oldAccessor, err := meta.Accessor(o)
```
nit: for consistency